api/provider/aws/common.go

```diff
@@ -247,29 +247,29 @@ func readSG(ec2api ec2iface.EC2API, groupName string) (*ec2.SecurityGroup, error
 }
 
 func readNewlyCreatedSG(ec2api ec2iface.EC2API, groupName string) (*ec2.SecurityGroup, error) {
+	var err error
 	var securityGroup *ec2.SecurityGroup
-	waitUntilSGisCreatedFN := func() (shouldRetry bool, err error) {
+
+	waitUntilSGisCreatedFN := func() (shouldRetry bool) {
 		securityGroup, err = readSG(ec2api, groupName)
 		if err != nil {
-			if err, ok := err.(awserr.Error); ok && strings.Contains(err.Error(), "does not exist") {
+			if serverError, ok := err.(awserr.Error); ok && strings.Contains(serverError.Error(), "does not exist") {
```

Comment:
using the name `serverError` is misleading because this is not a `errors.ServerError`, it is a `awserr.Error`.